src/documentInventory.ts

```diff
@@ -42,19 +42,29 @@ export class DocumentInventory {
   }
 
   constructor(public readonly document: IDocument<unknown>, protected resolver: IResolver) {
+    this.graph = null;
+    this.errors = null;
+
     const cacheKey = resolver instanceof Resolver ? resolver.uriCache : resolver;
     const cachedDocuments = DocumentInventory._cachedRemoteDocuments.get(cacheKey);
-    if (cachedDocuments) {
+    if (cachedDocuments !== void 0) {
       this.referencedDocuments = cachedDocuments;
     } else {
       this.referencedDocuments = {};
       DocumentInventory._cachedRemoteDocuments.set(cacheKey, this.referencedDocuments);
     }
   }
 
-  public async resolve() {
+  public async resolve(): Promise<void> {
+    if (!isObjectLike(this.document.data)) {
```

Comment:
@P0lip Looks like this would return `true` for an array. Is that something we should guard against?

cf. https://lodash.com/docs/4.17.15#isObjectLike

Reply:
Yeah, I was aware of that when introducing the code, but I thought that there was no strong reason for ignoring arrays.
I know that at the moment we primarily focus on API-related formats, yet in theory you can toss in an array either. I know "$ref" is heavily tied to API formats, but given we can handle arrays without any issues, I decided not to avoid them.
I verified the following input data, and we resolved the $ref just fine.
```json
[
  {
    "$ref": "#/1/foo"
  },
  {
    "foo": true
  }
]
```

What do you think?

Reply:
@P0lip Good enough for me. Maybe adding a test covering this passing use case?